src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.hpp

```diff
@@ -324,6 +324,32 @@ class CompositionalMultiphaseBase : public FlowSolverBase
    */
   void chopNegativeDensities( DomainPartition & domain );
 
+  /**
+   * @brief Utility function to keep the flow variables during a time step (used in poromechanics simulations)
+   * @param[in] keepFlowVariablesConstantDuringStep flag to tell the solver to freeze its primary variables during a time step
+   * @detail This function is meant to be called by a specific task before/after the initialization step
+   */
+  void keepFlowVariablesConstantDuringStep( bool const keepFlowVariablesConstantDuringStep )
+  { m_keepFlowVariablesConstantDuringStep = keepFlowVariablesConstantDuringStep; }
+
+  /**
+   * @brief Function to fix the initial state during the initialization step in coupled problems
+   * @param[in] time current time
+   * @param[in] dt time step
+   * @param[in] dofManager degree-of-freedom manager associated with the linear system
+   * @param[in] domain the domain
+   * @param[in] localMatrix local system matrix
+   * @param[in] localRhs local system right-hand side vector
+   * @detail This function is meant to be called when the flag m_keepFlowVariablesConstantDuringStep is on
+   *         The main use case is the initialization step in coupled problems during which we solve an elastic problem for a fixed pressure
+   */
+  void keepFlowVariablesConstantDuringStep( real64 const time,
```

Comment:
Here, unfortunately, we cannot make this function `private` or `protected` to compile on the GPU platforms because of the error:
```
The enclosing parent function ("keepFlowVariablesConstantDuringStep") for an extended __host__ __device__ lambda cannot have private or protected access within its class
```
We could move the body of `keepFlowVariablesConstantDuringStep` to a separate kernel file to overcome this issue, but this would be odd because `keepFlowVariablesConstantDuringStep` simply calls the `FieldSpecificationOps` utility functions...